langchain/agents/conversational_chat/base.py

```diff
@@ -98,11 +102,16 @@ def _construct_scratchpad(
             thoughts.append(human_message)
         return thoughts
 
+    @classmethod
+    def _validate_tools(cls, tools: Sequence[BaseStructuredTool]) -> None:
+        super()._validate_tools(tools)
+        validate_all_instance_of_base_tool(cls.__name__, tools)
```

Comment:
If we wanted to handle these at lint time, we could go back and redefine `Agent`, `BaseSingleActionAgent`, and then maybe do something similar for `BaseMultiActionAgent`, and then make a new agent executor as well, but this seemed like a fine compromise that avoided creating even more classes?